### PR TITLE
Add filestore test suite to ctest tool

### PR DESCRIPTION
**Pull request by gabrieljackson**
This change includes:
 - Refactor of ctest to support multiple configuration options and
   test types.
 - Switch to using logrus.
 - A ping healthcheck is now performed on all test installations.

Fixes https://mattermost.atlassian.net/browse/MM-30235

```release-note
Add filestore test suite to ctest tool
```

### CODE REVIEW

**Review by angeloskyratzakos — APPROVED** [review]
LGTM

**Review by Szymongib — APPROVED** [review]
Very nice tool, saves some manual testing 👍

File: cmd/tools/ctest/database.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+var databaseCmd = &cobra.Command{
+	Use:   "database",
+	Short: "Run the database test suite.",
+	RunE: func(command *cobra.Command, args []string) error {
+		port, _ := command.Flags().GetString("webhook-listener-port")
+
+		logger.Infof("Starting cloud webhook listener on port %s", port)
+
+		c := make(chan *model.WebhookPayload)
+		go databaseTests(command, c)
```
[the remark this answers is not in the document]
Comment (gabrieljackson) [reply cmd/tools/ctest/database.go:25]: Yeah, it waits and captures any more webhooks(which may be useful) until you kill the process, but I do plan to change it in the future.

File: cmd/tools/ctest/database.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+	installationDomain, _ := command.Flags().GetString("installation-domain")
+	version, _ := command.Flags().GetString("version")
+
+	databaseTypes := []string{
+		model.InstallationDatabaseMultiTenantRDSPostgres,
+		model.InstallationDatabaseMultiTenantRDSMySQL,
+		model.InstallationDatabaseSingleTenantRDSPostgres,
+		model.InstallationDatabaseSingleTenantRDSMySQL,
+		model.InstallationDatabaseMysqlOperator,
+	}
```
[the remark this answers is not in the document]
Comment (gabrieljackson) [reply cmd/tools/ctest/database.go:48]: Yep, also planned for the future :)

**Pull request merged**